feat: Add GenAI Experience Concierge sample app - agent design patterns demo

pablofgaeta opened the pull request:
# Description

Add the GenAI Experience Concierge agent design patterns demo to the sample apps directory. Covers the architecture for several design patterns: guardrail classification, function calling, semantic routing, and task planning (i.e. "Deep Research").

A deployed demo will be presented at Next '25 during the [Architecting Gen AI agents from LLM Ops to Advanced techniques
](https://cloud.withgoogle.com/next/25/session-library?session=SOL308&utm_source=copylink&utm_medium=unpaidsoc&utm_campaign=FY25-Q2-global-EXP106-physicalevent-er-next25-mc&utm_content=reg-is-live-next-homepage-social-share&utm_term=-) talk. We're hoping to make this publicly available for attendees to use.

- [x] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [x] You are listed as the author in your notebook or README file.
  - [x] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [x] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [x] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [x] Appropriate docs were updated (if necessary)

holtskinner commented:
Can you simplify this down at all? This is a massive application and I have a hard time following what to do with it?

Also, can you move it into the gemini/agents directory?

enriquekalven commented:
@holtskinner let us know if this is good to merge

holtskinner commented:
@pablofgaeta Thanks for your work to simplify the organization of this! Can you add more info to the main readme in your top directory to make it clear on what each directory is for and how they are to be used?

pablofgaeta commented:
> Can you add more info to the main readme in your top directory to make it clear on what each directory is for and how they are to be used?

@holtskinner Sure thing, just added a "Contents" section near the top of the readme to explain navigation and what each directory is for.

holtskinner commented:
Can you please remove the Parquet datasets from the PR?

I uploaded them to a public GCS bucket here: `gs://github-repo/generative-ai/gemini/agents/genai-experience-concierge/datasets` the file names are the same.
You can have the user import them into BigQuery from the public GCS bucket if possible.

pablofgaeta commented:
> Can you please remove the Parquet datasets from the PR?

@holtskinner Thanks for uploading them to GCS! I removed the parquet files and updated the docs and scripts to load from the bucket.
